appinventor/appengine/src/com/google/appinventor/client/explorer/folder/ProjectFolder.java

```diff
@@ -170,20 +169,27 @@ public void addProject(Project project) {
   }
 
   public void refresh() {
+    LOG.info("Refresh folder " + name);
+    if (nameLabel == null) {
+      LOG.info("Name Label is null ");
+    }
     nameLabel.setText(name);
+    LOG.info("Set date created ");
     dateCreatedLabel.setText(DATE_FORMAT.format(new Date(dateCreated)));
+    LOG.info("Set date modified ");
```

Comment:
We are doing a lot of logging here. We may want to revisit this or change the logging configuration in such a way that it won't spend much time in production.